data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -141,31 +142,106 @@ def getXValues(file):
     xSteps = [int(item/minExponent) for item in xSteps]
     return xSteps, minExponent
 
+def savePlot(name):
+    plt.tight_layout()
+    plt.savefig("{}/{}.png".format(pathToOutputData, name))
+    plt.close()
+
 def makeBarChart(data, minExponent, yTitleName, outputFilename):
-    chart = getBarChart(data)
+    prepareBarChart(data)
     setDescriptiveText(minExponent, yTitleName)
-    plt.tight_layout()
-    chart.savefig("{}/{}.png".format(pathToOutputData, outputFilename + "Bar"))
-    chart.close()
+    savePlot(outputFilename + "Bar")
 
 def makeLineChart(data, minExponent, yTitleName, outputFilename):
-    chart = getLineChart(data)
+    prepareLineChart(data)
     setDescriptiveText(minExponent, yTitleName)
-    plt.tight_layout()
-    chart.savefig("{}/{}.png".format(pathToOutputData, outputFilename + "Line"))
-    chart.close()
+    savePlot(outputFilename + "Line")
+
+def getMaxAndMinPercentage(data):
+    minVal = 999999999999
+    maxVal = -99999999999
+    for value in data.values():
+        percentages = list(value.values())
+        maxVal = max(maxVal, math.ceil(max(percentages))) #max av all kommande data
+        minVal = min(minVal, min(percentages))
+    return minVal, maxVal
+
+def setRelativePercentages(data, referenceValues):
+    for key in data.keys():
+        relativeValues = data[key].values()
+        percentages = [y/x for x, y in zip(referenceValues, relativeValues)] #Assume it's impossible for any run-time to be 0.
+        percentages = [ round(elem, 3) for elem in percentages ]
+        data[key].update(zip(data[key], percentages))
+
+def addRelativeLines(data, xTicks, colors):
+    counter = 1
+    for key in data.keys():
+        #print(data[key].values())
+        plt.semilogy(xTicks, data[key].values(), color=colors[counter], marker=markers[counter], base=2)
+        counter += 1
+
+def extractReferenceValuesAndXTicks(tmpData):
+    dataIter = iter(tmpData) 
+    firstKey = next(dataIter)
+    referenceValues = tmpData[firstKey].values()
+    xTicks = list(tmpData[firstKey].keys())
+    del tmpData[firstKey]
+    return referenceValues, xTicks
+
+def nextClosePowerOf2(x):  
+    if x >= 1:
+        x = int(x)
+        return 1 if x == 0 else 2**(x - 1).bit_length()
```

Comment:
Much cheaper (and less roundabout) to use left-shift in place of exponentiation: `1 << (x-1).bit_length()`.